src/backend/executor/nodeModifyTable.c

```diff
@@ -1571,6 +1576,17 @@ lreplace:;
 				{
 					TupleTableSlot *epqslot;
 
+					/*
+					 * TODO: DML node doesn't initialize `epqstate` parameter so
+					 * we exclude EPQ routine for this type of modification and
+					 * act as in RR and upper isolation levels.
+					 */
+					if (!epqstate)
+						ereport(ERROR,
```

Comment:
Can we move this check to the planning stage in ORCA planner and, if unsuccessful, fallback to the postgres planner?

Reply:
This case only occurs when there is a concurrent update. At the planning stage, we don't know if it will be concurrent or not.